packages/neon/neon/lib/src/widgets/app_bar.dart

```diff
@@ -77,19 +78,101 @@ class NeonAppBar extends StatelessWidget implements PreferredSizeWidget {
               ],
             ],
           ),
-          actions: [
-            const NotificationIconButton(),
-            IconButton(
-              onPressed: () {
-                AccountSettingsRoute(accountid: account.id).go(context);
-              },
-              tooltip: AppLocalizations.of(context).settingsAccount,
-              icon: NeonUserAvatar(
-                account: account,
+          actions: const [
+            NotificationIconButton(),
+            AccountSwitcherButton(),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+@internal
+class AccountSwitcherButton extends StatelessWidget {
+  const AccountSwitcherButton({
+    super.key,
+  });
+
+  Future<void> _onPressed(final BuildContext context) async {
+    final accountsBloc = Provider.of<AccountsBloc>(context, listen: false);
+    final accounts = accountsBloc.accounts.value;
+    final aa = accountsBloc.activeAccount.value!;
+
+    await showDialog(
+      context: context,
+      builder: (final context) {
+        final accountList = <Widget>[];
+
+        for (final account in accounts) {
+          if (account.id == aa.id) {
+            continue;
+          }
+
+          final tile = NeonAccountTile(
+            account: account,
+            onTap: () {
+              accountsBloc.setActiveAccount(account);
+              Navigator.of(context).pop();
+            },
+          );
+
+          accountList.add(tile);
+        }
+
+        final body = Column(
+          children: [
+            NeonAccountTile(
+              account: aa,
+              trailing: const Icon(Icons.check_circle),
+              onTap: Navigator.of(context).pop,
+            ),
+            const Divider(),
+            if (accountList.isNotEmpty)
+              SingleChildScrollView(
+                child: ListBody(children: accountList.toList()),
               ),
+            ListTile(
+              leading: const Icon(Icons.person_add),
+              title: Text(AppLocalizations.of(context).globalOptionsAccountsAdd),
+              onTap: () {
+                Navigator.pop(context);
+                const AddAccountRoute().push(context);
+              },
             ),
```

Comment:
This is duplicate with the new manage accounts page. I would remove it from here since it's just one more click away and you don't add accounts that often and it is quite clear where you can do that when the dialog is displayed.